keyvi/include/keyvi/dictionary/fsa/traversal/weighted_traversal.h

```diff
@@ -52,6 +52,20 @@ static bool WeightedTransitionCompare(const WeightedTransition& a, const Weighte
   return a.weight > b.weight;
 }
 
+template <>
+struct TraversalPayload<WeightedTransition> {
+  size_t current_depth;
+  uint32_t min_weight = 0;
+};
+
+template <>
+inline void TraversalState<WeightedTransition>::Add(uint64_t s, uint32_t weight, unsigned char l,
+                                                    TraversalPayload<WeightedTransition>* payload) {
+  if (weight >= payload->min_weight) {
```

Comment:
that's the only line causing "overhead" if traversal is done in sorted by weight order without filtering (`min_weight` alwasy `0`). I don't think this deservers another abstraction like we had before. It is rather unlikely that one wants to traverse by weight without filtering.